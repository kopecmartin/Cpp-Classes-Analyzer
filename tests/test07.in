class A {
    virtual int foo(void) =0;
};

class B : A {};

class C : A {
public:
    int var;
    virtual int foo(void);
};

class D : public C, B {
public:
    virtual int foo(void);
};
