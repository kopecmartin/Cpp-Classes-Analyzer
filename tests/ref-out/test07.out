<?xml version="1.0" encoding="UTF-8"?>
<class name="D" kind="concrete">
    <inheritance>
        <from name="C" privacy="public" />
        <from name="B" privacy="private" />
    </inheritance>
    <public>
        <attributes>
            <attribute name="var" type="int" scope="instance">
                <from name="C" />
            </attribute>
        </attributes>
        <methods>
            <method name="foo" type="int" scope="instance">
                <virtual pure="no" />
                <arguments></arguments>
            </method>
        </methods>
    </public>
</class>
